### PR TITLE
FXS

### PR DESCRIPTION
**Pull request by irischang020**

### CODE REVIEW

File: examples/scripts/ExampleMultipleChaperones.py [hunk 1]
```diff
@@ -1,64 +1,69 @@
 from __future__ import absolute_import
 import numpy as np
-from mpl_toolkits.mplot3d import Axes3D
 import matplotlib.pyplot as plt
+from matplotlib.colors import LogNorm
 import h5py as h5
 import time, os
 from pysingfel import *
```
Comment (chuckie82) [inline examples/scripts/ExampleMultipleChaperones.py:7]: remove. pysingfel imported twice

File: examples/scripts/ExampleMultipleChaperones.py [hunk 2]
```diff
@@ -1,64 +1,69 @@
...
     f.create_dataset('img', data=np_img)
     f.create_dataset('mask', data=np_mask, dtype=np.int16)
 
 fig = plt.figure(figsize=(10, 8))
 plt.subplot(131)
-plt.imshow(det.assemble_image_stack(patternOp),vmin=0, vmax=10)
-plt.title('Open state')
+plt.imshow(det.assemble_image_stack(patternOp),norm=LogNorm())
+plt.colorbar()
+plt.title('Open')
```
Comment (chuckie82) [inline examples/scripts/ExampleMultipleChaperones.py:58]: In title state whether it's photons or intensity

File: examples/scripts/ExampleMultipleChaperones.py [hunk 3]
```diff
@@ -1,64 +1,69 @@
...
-plt.title('Closed state')
+plt.imshow(det.assemble_image_stack(patternCl),norm=LogNorm())
+plt.colorbar()
+plt.title('Closed')
 plt.subplot(133)
-plt.imshow(det.assemble_image_stack(pattern),vmin=0, vmax=10)
-plt.title('%i Open + %i Closed'%(numOpen,numClosed))
+plt.imshow(det.assemble_image_stack(pattern),norm=LogNorm())
+plt.colorbar()
+plt.title('Mixed')
```
Comment (chuckie82) [inline examples/scripts/ExampleMultipleChaperones.py:66]: In title state how many open and how many closed

Comment (chuckie82) [reply examples/scripts/ExampleMultipleChaperones.py:66]: in a new subplot, imshow np_mask with title "polarization corrected region"

Comment (chuckie82) [reply examples/scripts/ExampleMultipleChaperones.py:66]: ok, that sounds good for now.
Later, we should introduce a way to assign a probability for each particle.

**Comment by chuckie82** [comment]
hmm.. somehow I don't see the changed code so unable to comment. I will try downloading this code.

File: examples/scripts/ExampleMultipleChaperones.py (hunk 3 above)
Comment (irischang020) [reply examples/scripts/ExampleMultipleChaperones.py:66]: We can assign a probability for each particle by passing in values to ratios. (default=None)
`experiment = ps.FXSExperiment(det, beam, particles, n_part_per_shot, ratios=None)`

**Comment by irischang020** [comment]
Pushed again just now.

File: examples/scripts/ExampleMultipleChaperones.py [hunk 4]
```diff
@@ -54,6 +54,7 @@
 plt.subplot(131)
 plt.imshow(det.assemble_image_stack(patternOp),norm=LogNorm())
 plt.colorbar()
+<<<<<<< HEAD
```
Comment (chuckie82) [inline examples/scripts/ExampleMultipleChaperones.py:57]: looks like these are not merged properly. is that true?

Comment (irischang020) [reply examples/scripts/ExampleMultipleChaperones.py]: The merge seems to have gone wrong somehow..pushed again.

**Comment by chuckie82** [comment]
Please commit as you make incremental changes like you are doing now.
Then double check and push everything at once when you are happy with it and want the reviewer to take a look.
otherwise, it's hard for the reviewer to know whether to start reviewing or not.

**Comment by irischang020** [comment]
Screen Shot 2020-07-10 at 12 15 57 PM

**Comment by irischang020** [comment]
All _fxs_ related functions in detector/base.py and reciprocal_detector.py no longer used are removed.
ExampleMultipleChaperones.py gets updated and new subplot added.

**Pull request merged**